### PR TITLE
Adds the security cyborg N.O.I.R. and warden upgrades as Jewvend-excusive rare loot.

### PR DESCRIPTION
**Pull request by jknpj**
wardenbro

Non-changelog changelog:
* Renames the check_icon() helper to has_icon() as check_icon() is used on rune code.
* Refactored hugborg's harm alarm to allow easy customization.
* Added Honkize() and Lawize() to harm alarms.
* Adds Noirmov, but no law board, just the datum to be used in the noir borg subtype and adminbus.
* Cleaned implanter code while adding the cyborg implanter and removed a lot of copypaste from it.
* The warden upgrade will search for icons that have a -H(heavy) or -warden variation and if found, apply them. Right now only the Noble icon has a Heavy variation.
* Both upgrades also work on hugborgs, giving them a special detective/warden overhaul.
* More unatomic changes that don't come to my mind right now.

:cl:
 * rscadd: Adds the security cyborg N.O.I.R. upgrade. It adds a tape recorder, evidence bag, cyborg zoom lens(works like binoculars), .38 revolver, cyborg .38 speedloader(charges one bullet every 10 seconds inside a cyborg recharger) and a worn-out gripper(can hold drinks, the detective's scanner and fingerprint cards). This upgrade cannot be researched.
 * rscadd: Adds the security cyborg warden upgrade. It adds a reinforced armour, battering ram, wrench, security code transmission device(locks/unlocks deployable barriers) and a cyborg implanter(recharges its implant after one minute inside a cyborg recharger). This upgrade cannot be researched.
 * rscadd: Adds both new upgrades to the vox vend premium list, giving traders a rare chance of finding them in their vend.

### CODE REVIEW

**Comment by Blithering** [comment]
just to make sure, this upgrades secborgs, not TGborgs?

**Comment by jknpj** [comment]
Both, actually

**Comment by ShiftyRail** [comment]
How about something to work better with cameras ?

File: code/__HELPERS/icons.dm [hunk 1]
```diff
@@ -141,7 +141,7 @@ proc/adjust_brightness(var/color, var/value)
 				return y_pixel - 1
 	return null
 
-/proc/check_icon(var/icon/I, var/wanted_state = null)
+/proc/has_icon(var/icon/I, var/wanted_state = null)
```
Comment (MadmanMartian) [inline code/__HELPERS/icons.dm:144]: But why?

Comment (DamianX) [reply code/__HELPERS/icons.dm:144]: unatomiiiiiiiiiiiiic

Comment (jknpj) [reply code/__HELPERS/icons.dm:144]: I-i can explain, check icon already exists in rune code: 

>/obj/effect/rune/proc/check_icon(var/mob/M = null)
    get_uristrune_cult(word1, word2, word3, M)

Comment (MadmanMartian) [reply code/__HELPERS/icons.dm:144]: That's even worse

**Comment by Durgen** [comment]
what happens if these get emagged?

**Comment by jknpj** [comment]
Nothing special

**Comment by SonixApache** [comment]
How hard would it be to give them the evidence forger / syndicuffs when emagged?

**Comment by Blithering** [comment]
infinite borg syndicuffs sounds like a recipe for doom

**Comment by Durgen** [comment]
how about explosive donuts

**Comment by jknpj** [comment]
I like the idea of donuts, i might tweak the cookie dispenser into joe/donut dispenser when using these upgrades

**Comment by jknpj** [comment]
Or maybe i can just remove the fucking cookie dispenser and give them a rapid service thing

**Pull request merged**